File: frontend/src/types/vendorAssignment.ts
export interface VendorAssignment {
  id: number;
  event_id: number;
  vendor_id: number;
  service: string;
  status: string;
}

export interface VendorAssignmentCreate {
  event_id: number;
  vendor_id: number;
  service: string;
  status?: string;
}

File: backend/app/models/vendor_assignment.py
from sqlalchemy import Column, Integer, String, ForeignKey
from app.database import Base

class VendorAssignment(Base):
    __tablename__ = "vendor_assignments"

    id = Column(Integer, primary_key=True, index=True)
    event_id = Column(Integer, ForeignKey("events.id"), nullable=False)
    vendor_id = Column(Integer, ForeignKey("vendors.id"), nullable=False)
    service = Column(String, nullable=False)
    status = Column(String, default="Assigned")